src/org/zaproxy/zap/extension/cspscanner/ContentSecurityPolicyScanner.java

```diff
@@ -0,0 +1,310 @@
+					Constant.messages.getString(MESSAGE_PREFIX + "xcsrf.desc"), xcspOptions.get(0),
+					cspHeaderFound ? Alert.RISK_INFO : Alert.RISK_LOW);
+		}
+
+		// X-WebKit-CSP is supported by Chrome 14+, and Safari 6+
+		Vector<String> xwkcspOptions = msg.getResponseHeader().getHeaders(HTTP_HEADER_WEBKIT_CSP);
+		if (xwkcspOptions != null && !xwkcspOptions.isEmpty()) {
+			raiseAlert(msg, Constant.messages.getString(MESSAGE_PREFIX + "xwkcsp.name"), id,
+					Constant.messages.getString(MESSAGE_PREFIX + "xwkcsp.desc"), xwkcspOptions.get(0),
+					cspHeaderFound ? Alert.RISK_INFO : Alert.RISK_LOW);
+		}
+
+		if (cspHeaderFound) {
+			ArrayList<Notice> notices = new ArrayList<>();
+			Origin origin = URI.parse(msg.getRequestHeader().getURI().toString());
+			String policyText = cspOptions.toString().replace("[", "").replace("]", "");
+ 			Policy pol = ParserWithLocation.parse(policyText, origin, notices);// Populate notices
+			
+			if (notices!= null && !notices.isEmpty()) {
+				raiseAlert(msg, Constant.messages.getString(MESSAGE_PREFIX + "notices.name"), id,
+						getCSPNoticesString(notices), getHeaderField(msg, HTTP_HEADER_CSP).get(0), getNoticesRisk());
+			}
+			
+			List<String> allowedWildcardSources = getAllowedWildcardSources(policyText, origin);
+			if (!allowedWildcardSources.isEmpty()) {
+				String allowedWildcardSrcs=allowedWildcardSources.toString().replace("[", "").replace("]", "");
+				String wildcardSrcDesc = MessageFormat
+						.format(Constant.messages.getString(MESSAGE_PREFIX + "wildcard.desc"), allowedWildcardSrcs);
+				raiseAlert(msg, Constant.messages.getString(MESSAGE_PREFIX + "wildcard.name"), id, wildcardSrcDesc,
+						getHeaderField(msg, HTTP_HEADER_CSP).get(0), Alert.RISK_MEDIUM);
+			}
+			
+			if (pol.allowsUnsafeInlineScript()) {
+				raiseAlert(msg, Constant.messages.getString(MESSAGE_PREFIX + "scriptsrc.unsafe.name"), id,
+						Constant.messages.getString(MESSAGE_PREFIX + "scriptsrc.unsafe.desc"),
+						getHeaderField(msg, HTTP_HEADER_CSP).get(0), Alert.RISK_MEDIUM);
+			}
+			
+			if (pol.allowsUnsafeInlineStyle()) {
+				raiseAlert(msg, Constant.messages.getString(MESSAGE_PREFIX + "stylesrc.unsafe.name"), id,
+						Constant.messages.getString(MESSAGE_PREFIX + "stylesrc.unsafe.desc"),
+						getHeaderField(msg, HTTP_HEADER_CSP).get(0), Alert.RISK_MEDIUM);
+			}
+		}
+
+		if (LOGGER.isDebugEnabled()) {
+			LOGGER.debug(
+					"\tScan of record " + String.valueOf(id) + " took " + (System.currentTimeMillis() - start) + " ms");
+		}
+	}
+
+	private void setNoticesRisk(int potentialRisk) {
+		if (potentialRisk > this.noticesRisk) {
+			this.noticesRisk = potentialRisk;
+		}
+	}
+	
+	private int getNoticesRisk() {
+		return noticesRisk;
+	}
+	
+	private String getCSPNoticesString(ArrayList<Notice> notices) {
+		char NEWLINE='\n';
+		StringBuilder returnSb = new StringBuilder();
+
+		ArrayList<Notice> errorsList = Notice.getAllErrors(notices);
+		if (!errorsList.isEmpty()) {
+			setNoticesRisk(Alert.RISK_MEDIUM);

```

Comment:
Been mulling this over. Im now thinking that making a weakness in a protective control a medium risk, without evidence that it can be leveraged negatively might be overstating things. Should I make errors low risk, then lump everything else into?

What about unsafe-inline, same thinking?